src/coreclr/jit/emitxarch.cpp

```diff
@@ -10722,6 +10722,20 @@ void emitter::emitDispInsHex(instrDesc* id, BYTE* code, size_t sz)
     }
 }
 
+// emitDispEmbBroadcastCount: Display the tag where embedded broadcast is activated to show how many elements are broadcasted.
+//
+// Arguments:
+//   id - The instruction descriptor
+//
+void emitter::emitDispEmbBroadcastCount(instrDesc* id)
+{
+    ssize_t baseSize = GetInputSizeInBytes(id);
+    id->idClearEvexbContext();
+    ssize_t vectorSize = (ssize_t)emitGetMemOpSize(id);
+    id->idSetEvexbContext();
```

Comment:
It seems like we shouldn't be altering the instrDesc during a debug-only print function. Ideally, this function would take a `const instrDesc*`.

Can you refactor `emitGetMemOpSize()` to be:

```
if (broadcast)
{
   return <broadcast result>
}
else
{
   return emitGetBaseMemOpSize(); // all the non-broadcast logic
}
```

Then, here:


```suggestion
    ssize_t vectorSize = (ssize_t)emitGetBaseMemOpSize(id);
```

Also, could this also do the Evexb check:

```
if (!id->idIsEvexbContext())
  return;

// otherwise, do the Evexb print logic
```

so the callers don't need to duplicate all the Evexb checking logic?